packages/admin-ui/src/masked-input/use-formatter.ts

```diff
@@ -0,0 +1,61 @@
+import { useMemo } from 'react'
+
+const MASK_INPUT_SYMBOL = '_'
+
+export interface UseFormatterProps {
+  mask: string
+  accept: RegExp
+}
+
+export function useFormatter(props: UseFormatterProps) {
+  const { mask, accept } = props
+  const formatter = useMemo(
+    () =>
+      createFormatter({
+        mask,
+        accept,
+      }),
+    [mask, accept]
+  )
+
+  return formatter
+}
+
+function createFormatter(props: UseFormatterProps) {
+  const { mask, accept } = props
+
+  function formatter(value: string) {
+    return value
+      .split('')
+      .map((char, i) => {
+        accept.lastIndex = 0
+
+        if (i > mask.length - 1) {
+          return ''
+        }
+
+        const maskChar = mask[i]
+        const nextMaskChar = mask[i + 1]
```

Comment:
I think this can return an error when `i === mask.length - 1`. I'm not sure if this breaks something, but isn't this the case of adding `mask?.[i+1]`?

Reply:
It's not necessary. This is no `java` or `c` to give you a pointer error, `nextMaskChar` will be undefined if the index does not exist - and this is okay. I check if this variable is `falsy` later on.